src/scripts/pokemons/PokemonList.ts

```diff
@@ -15865,7 +15865,10 @@ const pokemonList = createPokemonArray(
         'levelType': LevelType.mediumfast,
         'exp': 140,
         'catchRate': 120,
-        'evolutions': [new StoneEvolution('Charjabug', 'Vikavolt', GameConstants.StoneType.Thunder_stone)],
+        'evolutions': [
+            new StoneEvolution('Charjabug', 'Vikavolt', GameConstants.StoneType.Thunder_stone),
+            new DungeonRestrictedLevelEvolution('Vast Poni Canyon','Charjabug','Vikavolt', 20),
```

Comment:
So `Charjabug` can be evolved via either method?

Reply:
That's the idea. The stone is a Gen 8 method that was already there, I just added the Gen 7 "exposure to a special magnetic field" rule.

Reply:
I don't agree with having stone evolution as an option for Charjabug.

Reply:
While dex entry is restricted by region, I agree it should be like Leafeon and restricted to area, but when it stops being restricted I believe the stone should be fine.

Reply:
Looking at the reqs on Bulbapedia, I'll be re-doing that portion later today. Closing the pull request for now, I'll resubmit when it's done.